addon/globalPlugins/webAccess/ruleHandler/__init__.py

```diff
@@ -1298,55 +1380,30 @@ def setIfNotDefault(dic, key, value, default=None):
 		data = OrderedDict()
 		data["name"] = self.name
 		data["type"] = self.type
+		setIfNotDefault(data, "gestures", self.gestures, {})
 		if self.criteria:
 			items = data["criteria"] = []
 			for criteria in self.criteria:
 				items.append(criteria.dump())
-		setIfNotDefault(
-			data,
-			"mutation",
-			list(MUTATIONS.keys())[list(MUTATIONS.values()).index(self.mutation)] if self.mutation else None
-		)
-		setIfNotDefault(data, "gestures", self.gestures, {})
-		setIfNotDefault(data, "autoAction", self.autoAction)
-		setIfNotDefault(data, "multiple", self.multiple)
-		setIfNotDefault(data, "formMode", self.formMode, False)
-		setIfNotDefault(data, "skip", self.skip, False)
-		setIfNotDefault(data, "sayName", self.sayName, True)
-		setIfNotDefault(data, "customName", self.customName)
-		setIfNotDefault(data, "customValue", self.customValue)
-		setIfNotDefault(data, "comment", self.comment)
-
+		setIfNotDefault(data, "properties", self.properties.dump(), {})
 		return data
 
 	def load(self, data):
 		data = data.copy()
 		self.name = data.pop("name")
 		self.type = data.pop("type")
+		self.comment = data.pop("comment", None)
 		self.criteria = [Criteria(self, criteria) for criteria in data.pop("criteria", [])]
-		self.mutation = None
-		mutation = data.pop("mutation", None)
-		if mutation:
-			try:
-				self.mutation = MUTATIONS[mutation]
-			except LookupError:
-				log.exception((
-					"Unexpected mutation template id \"{mutation}\" "
-					"in rule \"{rule}\"."
-				).format(mutation=mutation, rule=self.name))
 		self.gestures = data.pop("gestures", {})
 		gesturesMap = {}
 		for gestureIdentifier in list(self.gestures.keys()):
 			gesturesMap[gestureIdentifier] = "notFound"
 		self.bindGestures(gesturesMap)
-		self.autoAction = data.pop("autoAction", None)
-		self.multiple = data.pop("multiple", False)
-		self.formMode = data.pop("formMode", False)
-		self.skip = data.pop("skip", False)
-		self.sayName = data.pop("sayName", True)
-		self.customName = data.pop("customName", None)
-		self.customValue = data.pop("customValue", None)
-		self.comment = data.pop("comment", None)
+		properties = data.pop("properties", {})
+		if not properties: # compatiiblity with old format
```

Comment:
This should go in the dataRecovery module.
There, I guess the `properties` key should be checked for existence, rather than checking the value for both absence and emptiness at the same time.